src/glyph.ts

```diff
@@ -5,20 +5,133 @@ import { Flow } from './tables';
 import { Element } from './element';
 import { BoundingBoxComputation } from './boundingboxcomputation';
 import { BoundingBox } from './boundingbox';
+import { Font } from './smufl';
+import { RenderContext } from './types/common';
+import { Stave } from './stave';
+
+export interface KeyProps {
+  stem_down_x_offset: number;
+  stem_up_x_offset: number;
+  key: string;
+  octave: number;
+  line: number;
+  int_value: number;
+  accidental: string;
+  code: string;
+  stroke: number;
+  shift_right: number;
+  displaced: boolean;
+}
+
+export interface TypeProps extends KeyProps {
+  getWidth(scale?: number): number;
+
+  code: string;
+  code_head: string;
+  stem: boolean;
+  rest: boolean;
+  flag: boolean;
+  stem_offset: number;
+  stem_up_extension: number;
+  stem_down_extension: number;
+  tabnote_stem_up_extension: number;
+  tabnote_stem_down_extension: number;
+  dot_shiftY: number;
+  line_above: number;
+  line_below: number;
+  beam_count: number;
+  code_flag_upstem: string;
+  code_flag_downstem: string;
+  position: string;
+}
+export interface DurationCode {
+  common: TypeProps;
+  type: Record<string, TypeProps>;
+}
 
-function processOutline(outline, originX, originY, scaleX, scaleY, outlineFns) {
-  let command;
-  let x;
-  let y;
+export interface GlyphProps {
+  code_head: string;
+  dot_shiftY: number;
+  position: string;
+  rest: boolean;
+  line_below: number;
+  line_above: number;
+  stem_up_extension: never;
```

Comment:
There's a lot of shared state with `TypeProps` -- why not inherit?

Reply:
There is a lot of shared but also differences. I would not go into that now, I would first integrate `getGlyphProps` from flow into this class. Wouldn´t you agree?

![image](https://user-images.githubusercontent.com/22865285/114593596-8e331900-9c8c-11eb-9528-59ea6aa1a152.png)